src/native/external/zlib-ng/deflate_fast.c

```diff
@@ -58,7 +58,9 @@ Z_INTERNAL block_state deflate_fast(deflate_state *s, int flush) {
         }
 
         if (match_len >= WANT_MIN_MATCH) {
-            check_match(s, s->strstart, s->match_start, match_len);
+            Assert(s->strstart <= UINT16_MAX, "strstart should fit in uint16_t");
```

Comment:
Why are we using zlib's custom `Assert` here and using the standard `assert()` in the `slide_hash` implementations? We should use the same type of assert throughout (likely `Assert`).

Reply:
`check_match` is a method that only gets defined in debug mode. So I figured it made sense to only call the debug-defined `Assert` here.

Reply:
Do you think I should always assert here? The problem was inside `check_match`